sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileMetadataStructSuite.scala

```diff
@@ -288,6 +288,20 @@ class FileMetadataStructSuite extends QueryTest with SharedSparkSession {
     )
   }
 
+  metadataColumnsTest("filter on metadata and user data", schema) { (df, _, f1) =>
```

Comment:
Does this test fail before?

Reply:
I think we should also check the physical plan and make sure the final file list is pruned.

Reply:
no, it won't fail before. I added this test just wanna make sure we only get metadata filters in `listFiles(...)` and do the correct filtering [here](https://github.com/apache/spark/pull/35055/files#diff-0d5b92aea54d993bd94249fdc9ba7d2616a3f4f04289ffaa2deb55a06585d4f9R78)

thanks for the suggestion! updated tests to check files!